src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size = 0;
+
+    public MyLinkedList() {
+        first = new Node<>();
+        last = new Node<>();
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first.value = value;
+            first.next = last;
+            size++;
+        } else if (size == 1) {
+            last.value = value;
+            first.next = last;
+            last.prev = first;
+            size++;
+        } else {
+            addNode(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index is not in range");
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> currentNode = getNode(index);
+            if (currentNode == first) {
+                Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+                currentNode.prev = newNode;
+                first = newNode;
+            } else {
+                Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+                currentNode.prev.next = newNode;
+                currentNode.prev = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            addNode(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index is not in range");
+        }
```

Comment:
same, let's make the exception message more informative. And lets move this index check to separate method as we use it several times in code